app/models/user.rb

```diff
@@ -8,4 +8,21 @@ class User < ApplicationRecord
   # The dependent: option allows to specify that associated records should be deleted when the owner is deleted
   # https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html >> Deleting from Associations
   has_many :notes, dependent: :delete_all
+
+  VALID_STATUS_LIST = %w[available working].freeze
+
+  validates :username, :email, presence: true
+  validates :current_status, inclusion: { in: VALID_STATUS_LIST }
+
+  after_initialize :init
+
+  attribute :current_status, :string, default: "available"
+
+  def init
+    self.username ||= email.split('@', 2)[0]
```

Comment:
Does this mean that admin@example.com and admin@test.de get the same username?
Is it ok that multiple users have the same username?

Reply:
Yes.
It was not specified that usernames should be unique.